cmd/snap-update-ns/utils_test.go

```diff
@@ -863,3 +863,104 @@ func (s *realSystemSuite) TestSecureOpenPathSymlinkedParent(c *C) {
 	c.Check(fd, Equals, -1)
 	c.Check(err, ErrorMatches, "not a directory")
 }
+
+func (s *realSystemSuite) TestPathIteratorEmpty(c *C) {
+	pi := update.NewPathIterator("")
+	c.Assert(pi.Path(), Equals, ".")
+
+	c.Assert(pi.Next(), Equals, true)
+	c.Assert(pi.PathSoFar(), Equals, ".")
+	c.Assert(pi.Segment(), Equals, ".")
+	c.Assert(pi.CleanSegment(), Equals, ".")
+
+	c.Assert(pi.Next(), Equals, false)
+}
+
+func (s *realSystemSuite) TestPathIteratorFilename(c *C) {
+	pi := update.NewPathIterator("foo")
+	c.Assert(pi.Path(), Equals, "foo")
+
+	c.Assert(pi.Next(), Equals, true)
+	c.Assert(pi.PathSoFar(), Equals, "foo")
+	c.Assert(pi.Segment(), Equals, "foo")
+	c.Assert(pi.CleanSegment(), Equals, "foo")
+
+	c.Assert(pi.Next(), Equals, false)
+}
+
+func (s *realSystemSuite) TestPathIteratorRelative(c *C) {
+	pi := update.NewPathIterator("foo/bar")
+	c.Assert(pi.Path(), Equals, "foo/bar")
+
+	c.Assert(pi.Next(), Equals, true)
+	c.Assert(pi.PathSoFar(), Equals, "foo/")
+	c.Assert(pi.Segment(), Equals, "foo/")
+	c.Assert(pi.CleanSegment(), Equals, "foo")
+
+	c.Assert(pi.Next(), Equals, true)
+	c.Assert(pi.PathSoFar(), Equals, "foo/bar")
+	c.Assert(pi.Segment(), Equals, "bar")
+	c.Assert(pi.CleanSegment(), Equals, "bar")
+
+	c.Assert(pi.Next(), Equals, false)
+}
+
+func (s *realSystemSuite) TestPathIteratorAbsolute(c *C) {
+	pi := update.NewPathIterator("/foo/bar")
```

Comment:
Could you please add similiar test for "/foo/bar/" (note the ending slash).

Reply:
Sure

Reply:
Done now